### PR TITLE
【Hackathon No.41】为 Paddle 新增 Rprop API -part

### PR DESCRIPTION
**Pull request by WintersMontagne10335**
### PR types

Others
### PR changes

APIs
### Description

为 Paddle 新增 Rprop API

#### Related links
- 黑客松：https://github.com/PaddlePaddle/Paddle/issues/57262
- RFC: https://github.com/PaddlePaddle/community/pull/746

#### Current progress
- [x] Python
- [x] C++
- [x] Unit testing passed
- [x] Complete unit testing cases
- [x] Add documents

### CODE REVIEW

**Comment by WintersMontagne10335** [comment]
@jeff41404 
Hello Jeff!
At present, the code structure has been completed and unit testing can run smoothly, with only some minor issues remaining to be fixed. If you have time, you can review it first. please don't stint your comments.
Wishing you a pleasant life！

File: python/paddle/optimizer/rprop.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    _learning_rates_acc_str = "learning_rates"
+
+    def __init__(
+        self,
+        learning_rate=0.001,
+        learning_rate_range=(1e-5, 50),
+        parameters=None,
+        etas=(0.5, 1.2),
+        grad_clip=None,
+        multi_precision=False,
```
Comment (jeff41404) [inline python/paddle/optimizer/rprop.py:107]: The parameter of `multi_precision` is not present in both docstring above and rfc. If it needs to be added, should modify docstring and rfc

Comment (WintersMontagne10335) [reply python/paddle/optimizer/rprop.py:107]: done.
- https://github.com/PaddlePaddle/community/pull/786

**Review by jeff41404 — DISMISSED** [review]
LGTM

File: python/paddle/optimizer/rprop.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            &\hspace{10mm} \Delta w_{ij}(t)=-sign(\frac{\partial E}{\partial w_{ij}}(t))*learning\_rate_{ij}(t)                                             \\
+            &\hspace{10mm} w_{ij}(t+1)=w_{ij}(t)+\Delta w_{ij}(t)                                                                                           \\
+            &\hspace{5mm} \}                                                                                                                                \\
+            &\hspace{0mm} \}                                                                                                                                \\
+       \end{aligned}
+
+    Parameters:
+        learning_rate (float|Tensor|LearningRateDecay, optional): The initial learning rate used to update ``Parameter``.
+            It can be a float value, a ``Tensor`` with a float type or a LearningRateDecay. The default value is 0.001.
+        learning_rate_range (tuple, optional): The range of learning rate.
```
Comment (sunzhongkai588) [inline python/paddle/optimizer/rprop.py:58]: `learning_rate_range` 和 `parameters` 加一下默认值的描述吧

Comment (WintersMontagne10335) [reply python/paddle/optimizer/rprop.py:58]: Done

**Comment by luotao1** [comment]
PR-CI-Codestyle-Check 没过

**Comment by WintersMontagne10335** [comment]
> PR-CI-Codestyle-Check 没过

是因为没拉取最新代码，已修正~~

**Review by sunzhongkai588 — APPROVED** [review]
LGTM

**Pull request merged**